common/src/main/java/io/netty/util/internal/NativeLibraryUtil.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+/**
+ * A Utility to Call the System.load() or System.loadLibrary().
+ */
+public final class NativeLibraryUtil {
+    public static void loadLibrary(String libName, boolean absolute) {
```

Comment:
Hi, @Scottmitch ! That was a good idea. But, the helper class's duty is to delegate the calling to System.load() and it will be injected into the native library's ClassLoader, then when its method is invoked, the native library would be loaded into the native library's ClassLoader, NOT the netty-common's ClassLoader.
So, **the helper class should be as simple as possible, and should not be a inner class, otherwise we have to load the outer class and its dependencies**.
But, the helper class can be package.